training/src/main/scala/com/airbnb/aerosolve/training/TrainingUtils.scala

```diff
@@ -258,6 +258,40 @@ object TrainingUtils {
       .collect
   }
 
+  def getLabelCounts(minCount : Int,
+                     input : RDD[Example],
+                     rankKey: String) : Array[((String, String), Int)] = {
+    input
+      .mapPartitions(partition => {
```

Comment:
seems mapPartitions is faster than map, we may want to change most of map into mapPartitions?